modules/mbedtls/Kconfig.tls

```diff
@@ -62,6 +62,9 @@ config MBEDTLS_KEY_EXCHANGE_ECDHE_PSK_ENABLED
 
 config MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED
 	bool "RSA-PSK based ciphersuite modes"
+	imply MBEDTLS_RSA_C if !PSA_CRYPTO_CLIENT
```

Comment:
Why not `select`? And what's the reasoning around `PSA_CRYPTO_CLIENT`?
+ IMO we could define a bundle that enables those 3 Kconfig option as this is done several times below.